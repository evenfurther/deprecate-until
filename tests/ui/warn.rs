#![deny(warnings)]

use deprecate_until::deprecate_until;

#[deprecate_until(remove = ">= 1.0")]
fn f1() {}

#[deprecate_until(remove = ">= 1.0", note = "Remove me")]
fn f2() {}

#[deprecate_until(remove = ">= 1.0", since = "1.0.0", note = "Remove me")]
fn f3() {}

fn main() {
    f1();
    f2();
    f3();
}
