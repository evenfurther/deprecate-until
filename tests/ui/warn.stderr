error: use of deprecated function `f1`: removal scheduled for version >= 1.0
  --> tests/ui/warn.rs:15:5
   |
15 |     f1();
   |     ^^
   |
note: the lint level is defined here
  --> tests/ui/warn.rs:1:9
   |
1  | #![deny(warnings)]
   |         ^^^^^^^^
   = note: `#[deny(deprecated)]` implied by `#[deny(warnings)]`

error: use of deprecated function `f2`: Remove me (removal scheduled for version >= 1.0)
  --> tests/ui/warn.rs:16:5
   |
16 |     f2();
   |     ^^

error: use of deprecated function `f3`: Remove me (removal scheduled for version >= 1.0)
  --> tests/ui/warn.rs:17:5
   |
17 |     f3();
   |     ^^
